### DISCUSSION TITLE
Does VIPS support writing interlaced GIFs

### DISCUSSION THREAD

**Discussion by richardsyeo** [libvips/libvips#3076] [Q&A]
Using Ubuntu 22.04 which appears to come with 8.12.1 of vips.

Does VIPS support writing interlaced GIFs?

Does this version of VIPS automatically write WEBP files that support incremental decoding?

**Comment by lovell** [comment]
Support for interlaced GIF encoding was added to libvips via https://github.com/libvips/libvips/pull/2984 and has not yet been included in a published version.

As far as I'm aware, incremental decoding of WebP images is independent of how they are encoded, but this is probably a question for the libwebp maintainers.

**Comment by jcupitt** [comment]
Sadly, libwebp does not really support incremental decode. You can supply the compressed image data in small chunks, but the decompressed pixels always appear in a huge lump, and only after all the compressed image data has been supplied. 

Google were talking about adding a tiled mode which would allow for large images and tile-wise decoding, but they lost interest and abandoned webp before implementing it. 

Perhaps webp2 (currently in development) will add this feature.

**Reply by jcupitt** [reply]
Actually, it looks like libwebp now has API for giving you just the part of the image that has been decoded so far:

https://docs.rs/libwebp-sys2/0.1.1/libwebp_sys/fn.WebPIDecGetRGB.html

libvips could probably use this to do incremental decode, though it wouldn't fix the 15k pixel limit.

**Reply by jcupitt** [reply]
See enhancement issue https://github.com/libvips/libvips/issues/3077
